src/firefly/js/ui/CompleteButton.jsx

```diff
@@ -121,8 +114,8 @@ export function CompleteButton ({onFail, onSuccess, groupKey=null, text='OK',
 CompleteButton.propTypes= {
     onFail: func,
     onSuccess: func,
-    groupsToUse: func,
-    groupKey: oneOfType([ string, arrayOf(string) ]),
+    // groupsToUse: func,                  // this is confusing and not necessary.  groupKey should be [string,func].
+    groupKey: oneOfType([ string, func ]),  // when func, it returns array of string
```

Comment:
I think this should be:
```js
groupKey: oneOfType([ string, arrayOf(string), func ])
```

Reply:
If the purpose of `func` is to only return `arrayOf(string)`, why complicate the api?  
`()=>(['abc', 'def'])` is  not hard to write and still very readable.
But, if you insist, I can change it.

Reply:
I did some checking we never use func as far as I can tell, but we do have an array of string. See below.

Reply:
I will change `groupKey` to `oneOfType([ string, arrayOf(string), func ])` as suggested.